subsys/net/ip/net_context.c

```diff
@@ -119,114 +120,103 @@ int net_context_get(sa_family_t family,
 {
 	int i, ret = -ENOENT;
 
-#if defined(CONFIG_NET_CONTEXT_CHECK)
-
-#if !defined(CONFIG_NET_IPV4)
-	if (family == AF_INET) {
-		NET_ASSERT_INFO(family != AF_INET, "IPv4 disabled");
-		return -EPFNOSUPPORT;
-	}
-#endif
-
-#if !defined(CONFIG_NET_IPV6)
-	if (family == AF_INET6) {
-		NET_ASSERT_INFO(family != AF_INET6, "IPv6 disabled");
-		return -EPFNOSUPPORT;
-	}
-#endif
-
-	if (!IS_ENABLED(CONFIG_NET_SOCKETS_CAN) && family == AF_CAN) {
-		NET_ASSERT_INFO((family != AF_CAN), "AF_CAN disabled");
-		return -EPFNOSUPPORT;
-	}
-
-	if (type == SOCK_RAW) {
-		if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) &&
-		    IS_ENABLED(CONFIG_NET_SOCKETS_CAN)) {
-			if (family != AF_PACKET && family != AF_CAN) {
-				return -EINVAL;
-			}
-		} else if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) &&
-			   !IS_ENABLED(CONFIG_NET_SOCKETS_CAN)) {
-			if (family != AF_PACKET) {
-				return -EINVAL;
+	if (IS_ENABLED(CONFIG_NET_CONTEXT_CHECK)) {
+		if (!IS_ENABLED(CONFIG_NET_IPV4) && family == AF_INET) {
+			NET_WARN("IPv4 disabled");
```

Comment:
Why did you remove the asserts and used logging instead? The point in the original code was that the error cases are exceptions and user will not trigger these 99% of the time, so the warning is too much in that case.

Reply:
sure it can be NET_DBG

The pb I had with the assert are:

- first the assert itself is a bit heavy as these are non-critical error (unlike a pointer that must be non null for instance. Let's say: the net_pkt pointer all through the recv path).
- second was readability: we do a test, and right below the same test, but reversed, is done for the sake of the assert. In some place it made quite a lot of lines just for the assert, generating a big amount of "useless" lines since we already did a proper test (it could have been a 'false' always put as first parameter to the net assert though)

Reply:
Lets use NET_DBG() then